crates/smtp_session/src/message.rs

```diff
@@ -0,0 +1,150 @@
+use error_handler::Error;
+
+#[derive(Debug)]
+pub struct SmtpMessage {
+    pub from: String,
+    pub to: Vec<String>,
+    pub subject: String,
+    pub body: String,
+}
+
+impl SmtpMessage {
+
+    pub fn new() -> SmtpMessageBuilder {
+        SmtpMessageBuilder::default()
+    }
+
+    pub fn to_imf(&self) -> String {
+        let mut imf_message = String::new();
+        
+        imf_message.push_str(&format!("From: {}\r\n", self.from));
+        
+        imf_message.push_str(&format!("To: {}\r\n", self.to.join(", ") ));
+
+        imf_message.push_str(&format!("Subject: {}\r\n", self.subject));
+        
+        imf_message.push_str("\r\n");
+        
+        imf_message.push_str(&self.body);
+        
+        imf_message
+    }
+}
+
+
+
+#[derive(Default)]
+pub struct SmtpMessageBuilder {
+    from: Option<String>,
+    to: Vec<String>,
+    subject: Option<String>,
+    body: Option<String>,
+}
+
+impl SmtpMessageBuilder {
+    pub fn new() -> Self {
```

Comment:
This could be renamed to default(). Is there any reason to not declare new(from, to, subject, body) and instead initialise fields with those small functions? You can just pass an empty string if some field is empty.

Reply:
As for now, there is really little reason to do so.
But in the future we may want to add more message fields
attached files, cc, bcc, replies, MIME multipart headers etc.
SmtpMessage::new(args) with 10+ args would definitely be a mess.
I believe a simple smtp_builder is not overengeneering, rather an
option to make it more scalable in the future.

Implementation was inspired with how one popular rust smtp client library handled this problem:
https://docs.rs/lettre/0.10.0-alpha.5/lettre/message/index.html#complex-mime-body

I will definitely change the naming though, to clarify what each method does.

Reply:
Thanks for clarifying!

Reply:
Thanks for clarifying!